frontend/src/metabase-types/api/mbql/references.ts

```diff
@@ -0,0 +1,40 @@
+export type ReferenceOptionsKeys =
+  | "source-field"
+  | "base-type"
+  | "join-alias"
+  | "temporal-unit"
+  | "binning";
+
+export type ReferenceOptions = Partial<
```

Comment:
With this approach it won't be possible to type options correctly. I believe we should use a regular type for it, e.g. 

```
interface ReferenceOptions {
  binning?: BinningOptions;
  "temporal-unit"?: TemporalUnit;
}

export type TemporalUnit =
  | "default"
  | "minute"
  | "minute-of-hour"
  | "hour"
  | "hour-of-day"
  | "day"
  | "day-of-week"
  | "day-of-month"
  | "day-of-year"
  | "week"
  | "week-of-year"
  | "month"
  | "month-of-year"
  | "quarter"
  | "quarter-of-year"
  | "year";


type BinningOptions = 
  | DefaultBinningOptions
  | NumBinsBinningOptions
  | BinWidthBinningOptions;

interface DefaultBinningOptions {
  strategy: "default";
}

interface NumBinsBinningOptions {
  strategy: "num-bins";
  "num-bins": number;
}

interface BinWidthBinningOptions {
   strategy: "bin-width"; 
  "bin-width": number;
}
```

search in `metabase-types/types` for other types

Reply:
fixed